docs/explanation/lifecycle-layer.rst

```diff
@@ -0,0 +1,198 @@
+
+From lifecycle to OCI layer
+===========================
+
+Rockcraft is a tool that creates OCI images using the same concepts and
+mechanisms that create snaps and charms: the lifecycle language from
+craft-parts. There is a significant difference between the works of the Craft
+lifecycle and the OCI specification, and one of Rockcraft's jobs is to bridge
+the gap between these two worlds. This page describes how this is accomplished.
+
+It is not necessary to know these details to use the tool effectively, but they
+might illuminate some concepts and help understand *why* the contents of a given
+rock are the way they are.
+
+Consider the following snippet of a ``rockcraft.yaml`` that creates a rock
+containing a bare-bones Python 3.10 interpreter:
+
+.. code-block:: yaml
+
+   # (...)
+   base: ubuntu@22.04
+
+   parts:
+     python-part:
+       plugin: nil
+       stage-packages:
+         - python3-minimal
+
+This rock has Ubuntu Jammy as its base, and includes ``python3-minimal``.
+Conceptually, this means that at build time craft-parts will pull in the
+``python3-minimal`` Ubuntu package and whatever dependencies it needs to work.
+Indeed, if we run ``rockcraft prime --shell-after``, we can see the final
+contents ready to be packed in the prime directory - this is the directory
+available at build-time through the ``${CRAFT_PRIME}`` environment variable:
+
+.. code-block:: bash
+
+   $ rockcraft prime --shell-after
+   $ cd ../prime
+   $ ls
+   bin  etc  lib  lib64  sbin  usr  var
+   $ ls usr/bin/
+   debconf               debconf-copydb          debconf-show  dpkg-divert              dpkg-realpath      dpkg-trigger  py3clean     python3
+   debconf-apt-progress  debconf-escape          dpkg          dpkg-maintscript-helper  dpkg-split         perl          py3compile   python3.10
+   debconf-communicate   debconf-set-selections  dpkg-deb      dpkg-query               dpkg-statoverride  perl5.34.0    py3versions  update-alternatives
+
+As we can see, the prime directory has the contents of the ``python3-minimal``
+package but also many of its dependencies, direct and otherwise.
+
+Pruning the ``prime`` directory
+-------------------------------
+
+Once the lifecycle is finished, Rockcraft packs the contents of the prime
+directory as a new OCI layer, directly as if the prime directory was the
+filesystem root ``/``. One consequence of the inclusion of a stage-package's
+dependencies is that the prime directory ends up having many files that the base
+Ubuntu layer already has. This can be seen, for example, by using a tool like
+``dive``:
```

Comment:
I think it's better to have a link instead: [Dive](https://github.com/wagoodman/dive)